frb_rust/src/handler.rs

```diff
@@ -209,26 +218,37 @@ impl<EH: ErrorHandler> Executor for ThreadPoolExecutor<EH> {
         TaskFn: FnOnce(TaskCallback) -> Result<TaskRet> + Send + UnwindSafe + 'static,
         TaskRet: IntoDart,
     {
-        const NUM_WORKERS: usize = 4;
-        lazy_static! {
-            static ref THREAD_POOL: Mutex<ThreadPool> = Mutex::new(ThreadPool::with_name(
-                "frb_executor".to_string(),
-                NUM_WORKERS
-            ));
-        }
-
         let eh = self.error_handler;
         let eh2 = self.error_handler;
-        THREAD_POOL.lock().execute(move || {
-            let wrap_info2 = wrap_info.clone();
+
+        let WrapInfo { port, mode, .. } = wrap_info;
+
+        // BroadcastChannel can't actually cross the thread border since
+        // it is not a transferable, but we can carry its name instead.
+        #[cfg(wasm)]
+        let (port, channel) = if let Some(channel) = port
+            .as_deref()
+            .and_then(wasm_bindgen::JsCast::dyn_ref::<web_sys::BroadcastChannel>)
+        {
+            (None, Some(crate::rust2dart::Channel(channel.name())))
+        } else {
+            (port, None)
+        };
+
+        spawn!(|port: Option<MessagePort>| {
+            #[cfg(wasm)]
+            let port = port.or_else(|| channel.as_ref().map(crate::rust2dart::Channel::port));
```

Comment:
If I understand correctly, here we are doing: If wasm, let the name cross thread bound; if io, just directly move things.

Then, what about the suggestion I mentioned above: We can create something like `SendableMessagePort`, which *encapsulates* the logic here. Such that, we can always send `SendableMessagePort` freely.

Reply:
I tried to do something like this, but quickly realized how unrealistic it is:
- `Send` is a trait that must be unsafely implemented, i.e. it creates additional overhead for us to uphold the constraints manually.
- Types like `ZeroCopy<Vec<u8>>` are already `Send`, but we might want to "transfer" this type in the future as well. For this reason I consider `Send` and transferability to be separate concepts, and if `ZeroCopy` is to be reimplemented in the future a trait such as `Transfer` may be needed. This way, the macro can depend on `Transfer` being implemented instead of `Send`.

Reply:
> Send is a trait that must be unsafely implemented

IMHO it will be *auto* implemented, so we should *not* implement it by ourselves, but should instead ensure the data struct satisfy the requriement.

In this specific code, seems that we are not dealing with ZeroCopy, but instead dealing with "how can we get port across threads". Thus, we should extract a new struct to encapsulate.

Reply:
> If I understand correctly, here we are doing: If wasm, let the name cross thread bound; if io, just directly move things.

You might be thinking of `BroadcastChannel`; this is passing a `MessagePort` as-is as one of the transferables.